src/controllers/users.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+const userServise = require('../services/users.js');
+
+let nextId = 0;
+
+function add(req, res) {
+  const { name } = req.body;
+
+  if (!name) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const newUser = {
+    id: nextId++,
+    name,
+  };
+
+  userServise.add(newUser);
+
+  res.statusCode = 201;
+  res.send(newUser);
+};
+
+function getAll(req, res) {
+  const users = userServise.getAll();
+
+  res.statusCode = 200;
+
+  if (!users) {
+    res.send([]);
+
+    return;
+  }
+
+  res.send(users);
+};
+
+function getById(req, res) {
+  const userId = Number(req.params.id);
+
+  if (typeof userId !== 'number') {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const foundUser = userServise.findById(userId);
+
+  if (!foundUser) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.statusCode = 200;
+
+  res.send(foundUser);
+};
+
+function update(req, res) {
+  const userId = Number(req.params.id);
```

Comment:
Try this way 👇 
```suggestion
  const { userId: Number(userId)  } = req.params;
```

Using destructure - is a better option here. And in future test tasks when it possible.